src/Api.js

```diff
@@ -89,13 +92,68 @@ class Api {
 	}
 
 	/**
-	 * Get the WhoColor data for a given wiki page.
+	 * Get the WikiWho data for a given wiki page. Use onError and onResults to set return handlers.
 	 *
-	 * @param  {string} wikiUrl URL of the wiki page that we want to analyze.
-	 * @return {Promise}
+	 * @param {string} wikiUrl URL of the wiki page that we want to analyze.
 	 */
 	getData( wikiUrl ) {
-		return $.getJSON( this.getAjaxURL( wikiUrl ) );
+		$.getJSON( this.getAjaxURL( wikiUrl ) ).then( ( function ( result ) {
+			// Handle error.
+			if ( !result.success && result.info && this.errorHandler ) {
+				// Turn the English error message into a i18n message key.
+				// @TODO Remove this when the WikiWho API returns useful error codes.
+				let errorCode = result.info
+					.toLowerCase()
+					.replace( / /g, '-' )
+					.replace( /[^a-z-]/g, '' );
+				// Possible messages:
+				// eslint-disable-next-line
+				// ext-whowrotethat-requested-data-is-not-currently-available-in-wikiwho-database-it-will-be-available-soon
+				this.errorHandler( 'ext-whowrotethat-' + errorCode );
+				return;
+			}
+
+			let cleanedResults = {
+				replacementHtml: result.extended_html,
+				users: {},
+				revisionsByUser: {}
+			};
+
+			// Reorganise the revisions to group by user ID.
+			// revisions structure: {rev_id: [timestamp, parent_rev_id, user_id, editor_name ], ...}
+			for ( let [ revId, rev ] of Object.entries( result.revisions ) ) {
+				let userId = rev[ 2 ];
+				if ( cleanedResults.revisionsByUser[ userId ] === undefined ) {
+					cleanedResults.revisionsByUser[ userId ] = {};
+				}
+				cleanedResults.revisionsByUser[ userId ][ revId ] = {
+					revId: revId,
+					timestamp: rev[ 0 ],
+					parentRevId: rev[ 1 ],
+					userId: rev[ 2 ]
+				};
+			}
+
+			// Round the authorship percentages.
+			// present_editors structure: [ [ username, user_id, score ], ... ]
+			result.present_editors.forEach( function ( user ) {
+				cleanedResults.users[ user[ 1 ] ] = {
+					username: user[ 0 ],
+					userId: user[ 1 ],
+					score: parseFloat( user[ 2 ] ).toFixed( 1 )
+				};
+			} );
+
+			this.resultHandler( cleanedResults );
+		} ).bind( this ) );
+	}
+
+	onError( callback ) {
+		this.errorHandler = callback;
```

Comment:
This has a bit of a code-smell of allowing (or needing) an external thing to deal with internal workings of how the API (and errors) are returned. See my comment above regarding returning the promise in `.getData`

Same for `onResult`